core/resources.rs

```diff
@@ -64,6 +64,10 @@ pub trait Resource: Any + 'static {
   fn backing_fd(self: Rc<Self>) -> Option<std::os::unix::prelude::RawFd> {
     None
   }
+
+  fn size_hint(&self) -> (u64, Option<u64>) {
+    (64 * 1024, None)
```

Comment:
```suggestion
    (0, None)
```

There is no guarantee that the resource is at least 64 KB. The only constraint that matches all resources is min=0 max=infinite, which is `(0, None)`

Reply:
My thought was the minimum buffer size to use for reading instead of the actual content size. You're right though, it's the content size hint so it should be zero.